### PR TITLE
New logger interface and implementation

### PR DESCRIPTION
**Pull request by tsjohns9**
Created a new logging interface and struct that acts like the Logrus FieldLogger, and handles logp writing like we do currently so that additional fields can be added to log messages

### CODE REVIEW

File: pkg/util/log/logger.go [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+// WithError adds an error field to the message
+func (l *logger) WithError(err error) FieldLogger {
+	return &logger{entry: l.entry.WithError(err)}
+}
+
+// Debugf prints a formatted debug message
+func (l *logger) Debugf(format string, args ...interface{}) {
+	if l.isLogP() {
+		args = append(args, l.formatEntries()...)
+		logp.L().Debugf(format, args)
```
Comment (tsjohns9) [inline pkg/util/log/logger.go:78]: @jcollins-axway The original log package uses a const called `debugSelector` when writing debug logs to logp. I don't know exactly what we are trying to do with that. Do you mind updating this PR with anything else we may need to keep it consistent with the current implementation?

File: pkg/util/log/logger.go [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+	Infof(format string, args ...interface{})
+	Infoln(args ...interface{})
+
+	Warn(args ...interface{})
+	Warnf(format string, args ...interface{})
+	Warnln(args ...interface{})
+}
+
+// Redactor interface for redacting log messages
+type Redactor interface {
```
Comment (tsjohns9) [inline pkg/util/log/logger.go:40]: Should the `Redactor` be added to the `FieldLogger` interface?

File: pkg/util/log/logger.go (hunk 1 above)
Comment (jcollins-axway) [reply pkg/util/log/logger.go:78]: Logp does not have trace, so to differ the 2 logging events the debugSelector is used to name the logger in logp.

File: pkg/util/log/logger.go (hunk 2 above)
Comment (jcollins-axway) [reply pkg/util/log/logger.go:40]: The AMPGW team added these, would want to check to see how they are using this.

**Review by jcollins-axway — DISMISSED** [review]

**Comment by tsjohns9** [comment]
I updated a few components to use the FieldLogger, like the jobs, stream, cache manager, and client

**Review by jcollins-axway — DISMISSED** [review]

**Review by sbolosan — DISMISSED** [review]

**Review by sbolosan — APPROVED** [review]

File: pkg/transaction/metric/metricscollector.go [hunk 3]
```diff
@@ -161,8 +166,17 @@ func (c *collector) Execute() error {
 	c.usageEndTime = now()
 	c.metricEndTime = now()
 	c.orgGUID = c.getOrgGUID()
-	log.Debugf("Generating usage event [start timestamp: %d, end timestamp: %d]", util.ConvertTimeToMillis(c.usageStartTime), util.ConvertTimeToMillis(c.usageEndTime))
-	log.Debugf("Generating metric event [start timestamp: %d, end timestamp: %d]", util.ConvertTimeToMillis(c.metricStartTime), util.ConvertTimeToMillis(c.metricEndTime))
+	c.logger.
+		WithField("start timestamp", util.ConvertTimeToMillis(c.usageStartTime)).
```
Comment (jcollins-axway) [inline pkg/transaction/metric/metricscollector.go:170]: I know we have this in other places, but wondering if we should skip the spaces for the field names.  I know json will still be valid, just thinking we may want to make it a standard?

**Review by jcollins-axway — APPROVED** [review]

**Pull request merged**